### DISCUSSION TITLE
(Webpack) Why is versioning disabled in dev by default?

### DISCUSSION THREAD

**Discussion by php4fan** [symfony/symfony#44161] [Q&A] [answered]
I'm new to Symfony (and to Webpack Encore).
I created a simple project following the great tutorials at SymfonyCasts. Among other things I installed Webpack Encore, Bootstrap, and I started organizing my app's css files.

After the first trivial edit to my css file, I ran into a stupid problem: I rebuilt the compiled CSS, opened my app in my browser served from Symfony's local server, and the css file that I was seeing was the unedited old one. After checking that I hadn't forgot to rebuild the css, I cleared the broswer's cache and the edits came online.
So, I was like: **why the heck isn't the file being versioned out of the box?**

I looked up how you enabled versioning, went to edit **webpack.config.js** to enable versioning thinking to myself _"Isn't it a bit stupid that it's not enabled by default? Whatever, maybe this is part of the 'start small' phylosophy..."_ and then found this:

```
Encore
    .setOutputPath('public/build/')
    .setPublicPath('/build')
    //....
    .enableVersioning(Encore.isProduction())  /// <<<<<<<<<<< THIS!!!!
```

So, it's not like it's not enabled by default. It's **purposefully enabled only on Production** by default.

Before I report this as a bug, what is the reasoning behind that that I might be missing? Why would you expect to not need versioning in development? What part of a good, best-practice development workflow am I not applying that would make versioning not necessary in development?

**Comment by weaverryan** [comment, marked as answer]
Hey @php4fan!

Hmm, the line in question in webpack.config.js was probably code that I originally created. I think the reason it wasn't enable on production are historical, and really don't apply anymore.

In the first version of Encore, there was no `encore_entry_script_tags()`. Instead, you just added `<script src="{{ asset('build/app.js')">` to your template. In this situation, versioning only worked if you had your `json_manifest_path` configured correctly... which you probably did, but we didn't have any guarantee. And since versioning was the ONLY feature that truly required you to have this setup correctly, we made it an "extra" feature for production (we may even have commented it out originally, I can't remember).

Anyways, now that you effectively NEED to use `encore_entry_script_tags()` with Encore (due to Webpack's code splitting... which you CAN disable, but I don't think anyone does), I can't think of a reason why this should NOT be used in dev. In other words, I agree with you :). The change would be made here: https://github.com/symfony/recipes/blob/a624d3bf1def68bbd2d542b3a99216c0af090552/symfony/webpack-encore-bundle/1.9/webpack.config.js#L46

**Reply by php4fan** [reply]
> I think the reason it wasn't enable on production are historical

You mean in dev, right?

**Reply by stof** [reply]
@weaverryan is the manifest properly regenerated in watch mode now ? IIRC, we used to have issues in the past (but that was before the webpack 5 migration so may be resolved now) where the manifest was not updated when watching, and so versioned files were becoming wrong after the watcher was regenerating some files.
